From 34dd0e95665abdf2ca37f1c603668ad1c1cdc5a8 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Tue, 23 Jul 2019 11:02:25 +0800
Subject: [PATCH 1419/1494] drm/i915: Remove i915 ggtt WA since GT E

The APL already fixed this bug since GT E, bug exist from
0 to D, apply WA according GT stepping.

Change-Id: If46a0e86c1dc3bfd8dfa3c45f2fe4aea018f688b
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 6d56078..452c8d1 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2476,6 +2476,8 @@ intel_info(const struct drm_i915_private *dev_priv)
 #define BXT_REVID_B0		0x3
 #define BXT_REVID_B_LAST	0x8
 #define BXT_REVID_C0		0x9
+#define BXT_REVID_D0		0xC
+#define BXT_REVID_E0		0xD
 
 #define IS_BXT_REVID(dev_priv, since, until) \
 	(IS_BROXTON(dev_priv) && IS_REVID(dev_priv, since, until))
@@ -2724,7 +2726,7 @@ static inline bool intel_scanout_needs_vtd_wa(struct drm_i915_private *dev_priv)
 static inline bool
 intel_ggtt_update_needs_vtd_wa(struct drm_i915_private *dev_priv)
 {
-	return IS_BROXTON(dev_priv) && intel_vtd_active();
+	return IS_BXT_REVID(dev_priv, 0, BXT_REVID_D0) && intel_vtd_active();
 }
 
 int intel_sanitize_enable_ppgtt(struct drm_i915_private *dev_priv,
-- 
2.7.4

